File: RXSDK/RXSDK-Android/rxsdk_base/src/main/java/com/ruixue/passport/LoginMethod.java
package com.ruixue.passport;

import android.text.TextUtils;

import androidx.annotation.NonNull;

import com.ruixue.openapi.RXGlobalData;
import com.ruixue.utils.ResUtils;

import java.util.HashSet;

public class LoginMethod {
    public static final String MORE = "more";
    /**
     * 游客登录:
     * method: guest
     */
    public static final String GUEST = "guest";
    /**
     * appid
     */
    public static final String WECHAT = "wechat";
    /**
     * 账号密码登录
     * method: username
     */
    public static final String USERNAME = "username";
    /**
     * 手机一键登录
     * method:quickphone
     * ext:
     * access_token
     */
    public static final String QUICKPHONE = "quickphone";

    /**
     * 手机/邮箱验证码登录，username 可为手机号或者邮箱地址
     */
    public static final String CAPTCHACODE = "captchacode";

    /**
     * /应用宝SDK登录:
     * //method: ysdk
     * //ext:
     * //    form
     * //    openid
     * //    avatar
     * //    nickname
     */
    public static final String YSDK = "ysdk";
    /**
     * //快手登录
     * //method: kuaishou
     * //ext:
     * //    game_id
     * //    game_token
     */
    public static final String KUAISHOU = "kuaishou";
    /**
     * 百度网讯
     * method:baidunet
     * ext:
     * access_token  # 百度返回的用户token
     */
    public static final String BAIDUNET = "baidunet";
    /**
     * 华为
     * method:huawei
     * ext:
     * playerId  # 华为返回的用户id
     * displayName # 华为返回的显示名称
     */
    public static final String HUAWEI = "huawei";

    public static final String HONOR = "hihonor";
    /**
     * 华为
     * method:hwjos
     * ext:
     * playerId  # 华为返回的用户id
     * openid # 华为返回的openid
     * displayName # 华为返回的displayName
     * playerLevel # 华为返回的playerLevel
     * playerSign # 华为返回的playerSign
     * ts # 华为返回的ts
     */
    public static final String HWJOS = "hwjos";
    /**
     * 小米
     * method:mi
     * ext:
     * session # 小米返回的用户会话session
     * uid # 小米返回的用户id
     * nickname # 小米返回的nickname昵称
     */
    public static final String MI = "mi";
    /**
     * OPPO
     * method:oppo
     * ext:
     * token # oppo返回的token
     * openid # oppo返回的openid
     * OPPO小游戏
     * method:oppomini
     * ext:
     * token # oppo小游戏返回的用户token
     */
    public static final String OPPO = "oppo";

    /**
     * OPPO 海外
     */
    public static final String OPPO_OS = "oppo_os";
    /**
     * VIVO
     * method:vivo
     * ext:
     * authToken # vivo返回的用户token
     * username # vivo返回的username用户名
     */
    public static final String VIVO = "vivo";
    public static final String XUTENG = "xuteng";
    public static final String HUYA = "huya";
    /**
     * 抖音联运
     * method:douyin
     * ext:
     * access_token # 抖音用户access_token
     */
    public static final String DOUYIN = "douyin";
    /**
     * 虚拟账号登录（用于未接入通行证但又想使用瑞雪其他接口的场景）
     * method:virtual
     * ext:
     * logindata # (必须)登录凭证，由 ruixuego 服务端 SDK 的 GenerateVirtualLoginData 接口生成
     * nickname # (非必须)用户昵称
     * avatar # (非必须)用户头像
     * sex # (非必须)性别,1:男,0:女.(必须是字符串类型的值)
     * regtime # (必须)注册时间，格式为 YYYY-MM-DD HH:ii:ss，例：2022-01-02 09:10:11
     */
    public static final String VIRTUAL = "virtual";

    public static final String BILIBILI = "bilibili";
    public static final String M4399 = "4399";

    public static final String TAPTAP = "taptap";

    public static final String GOOGLE = "google";

    public static final String FACEBOOK = "facebook";

    public static final String LINE = "line";

    public static final String ZALO = "zalo";

    public static final String TIKTOK = "tiktok";

    public static final String QOO = "qoo";

    public static final String SNAPCHAT = "snapchat";

    public static final String INSTAGRAM = "instagram";

    public static final String REDDIT = "reddit";

    public static final String MUMU = "mumu";

    public static final String M9GAME = "jiuyou";

    public static final String LEIDIAN = "leidian";

    public static final String M007 = "client_007";

    public static final String QUICK = "quick";
    public static final String WEIZHI = "weizhi";

    public static final String VK = "vk";

    private static final HashSet<String> OWN_LOGIN = new HashSet<>();

    static {
        OWN_LOGIN.add(VIRTUAL);
        OWN_LOGIN.add(GUEST);
        OWN_LOGIN.add(USERNAME);
        OWN_LOGIN.add(CAPTCHACODE);
    }


    private final String method;
    private final String name;
    private final int icon;
    private String ext = "";

    public String getMethod() {
        return method;
    }

    public String getName() {
        return TextUtils.isEmpty(name) ? method : name;
    }

    public int getIcon() {
        return icon;
    }

    public LoginMethod(@NonNull String method, String ext) {
        this.method = method;
        if (GUEST.equals(this.method) && !TextUtils.isEmpty(RXGlobalData.getGuestTitle())) {
            this.name = RXGlobalData.getGuestTitle();
        } else {
            this.name = ResUtils.getInstance().getString("method_" + method + ext);
        }
        this.icon = ResUtils.getInstance().getDrawableId("rx_ico_" + method + ext);
    }

    public static String getName(String method) {
        return new LoginMethod(method, "").getName();
    }

    public static LoginMethod create(String method) {
        return new LoginMethod(method, "");
    }

    public static LoginMethod create(String method, String ext) {
        return new LoginMethod(method, ext);
    }

    public static boolean isChannel(String method) {
        return method != null && !OWN_LOGIN.contains(method);
    }


}

File: RXSDK/RXSDK-Unity/Packages/com.ruixue.unitysdk.base/Runtime/Common/LoginMethod.cs
namespace RuiXue
{
    public static class LoginMethod
    {
        public const string Virtual = "virtual";// 虚拟账号。用于一些暂时不想接入瑞雪通行证的账号体系，但要使用瑞雪的分享、统计等功能的游戏项目，相当于将游戏项目自身作为一个三方账号平台。
        public const string CaptchaCode = "captchacode"; // 手机/邮箱验证码登录，username 可为手机号或者邮箱地址
        public const string Guest = "guest"; // 游客账号，以设备信息生成设备码作为登录凭证。
        public const string Username = "username"; // 普通账号，包括手机账号、邮箱账号和自定义用户名的账号
        public const string QuickPhone = "quickphone"; // 阿里一键登录，即将当前手机号作为用户名。
        public const string Wechat = "wechat"; // 通过微信 APP 授权登录微信账号
        public const string Minigame = "minigame"; // 通过微信小游戏登录微信账号
        public const string MobileQQ = "mobileqq"; // QQ 账号登录。
        public const string YSDK = "ysdk"; // 应用宝账号登录。
        public const string KuaiShou = "kuaishou"; // 登录快手账号。 
        public const string Apple = "apple"; // 登录苹果账号。
        public const string BaiduNet = "baidunet"; // 登录百度网讯账号
        public const string HuaWei = "huawei"; // 登录华为账号。
        public const string HWJos = "hwjos"; // 登录华为拂袖账号
        public const string HuaWeiH5 = "huaweih5"; // 通过 H5 的方式登录华为账号
        public const string Mi = "mi"; // 登录小米账号。
        public const string Vivo = "vivo"; // 登录 vivo 账号。
        public const string Oppo = "oppo"; // 登录 oppo 账号。
        public const string MeiTuan = "meituan"; // 登录美团账号
        public const string DouYin = "douyin"; // 登录抖音账号。
        public const string DouYinH5 = "douyinh5"; // 登录抖音小游戏账号。
        public const string FaceBook = "facebook"; // 登录 facebook 账号。
        public const string Google = "google"; // 登录 google 账号。
        public const string TapTap = "taptap"; // 登录 TapTap 账号。
        public const string BiliBili = "bilibili"; // 登录 bilibili 账号
        public const string Line = "line"; // 登录 LINE 账号
        public const string _4399 = "4399"; // 登录 4399 账号
        public const string WeiLe = "weile"; // 登录微乐小游戏账号
        public const string JiXiang = "jixiang"; // 登录吉祥小游戏账号。
        public const string XinYue = "xinyue";// 登录心悦小游戏账号。
        public const string Qoo = "qoo";// Qoo渠道登录
        public const string Zalo = "zalo";// zalo登录
        public const string Tiktok = "tiktok";// tiktok登录
        public const string SnapChat = "snapchat";//snapchat登录，snapChat无登录功能
        public const string Instagram = "instagram";//instagram登录
        public const string Reddit = "reddit";// Reddit登录
        public const string Default = "default"; //自定义登录
        public const string LEIDIAN = "leidian";// 雷电
        public const string HIHONOR = "hihonor";// 荣耀
        public const string MuMu = "mumu";// 网易 MuMu/Yofun 渠道登录
        public const string Huya = "huya";// 虎牙联运渠道登录
        public const string Xuteng = "xuteng";// 栩腾渠道登录
        public const string QUICK = "quick";// quick
    }
}